cdap-app-fabric/src/main/java/co/cask/cdap/metadata/profile/ProfileMetadataMessageProcessor.java

```diff
@@ -265,4 +250,26 @@ private ProfileId getResolvedProfileId(EntityId entityId) {
     }
     return result;
   }
+
+  /**
+   * Get the set of programTypes allowed to use compute profiles.
+   */
+  private Set<ProgramType> getProfileAllowedProgramTypes() {
```

Comment:
this looks like it can be changed to a private static final class variable instead of calculated each time.

Reply:
I meant changing it to a private static final variable instead of having this method.

Reply:
oh gotcha, sorry read too quickly